ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -91,7 +91,7 @@ protected void endTransaction(final TransactionManager tm, final Transaction tx)
                 tm.commit();
             } else if (txStatus == Status.STATUS_MARKED_ROLLBACK) {
                 tm.rollback();
-            } else if (txStatus == Status.STATUS_ROLLEDBACK) {
+            } else if (txStatus == Status.STATUS_ROLLEDBACK || txStatus == Status.STATUS_ROLLING_BACK) {
                 // handle reaper canceled (rolled back) tx case (see WFLY-1346)
                 // clear current tx state and throw RollbackException (EJBTransactionRolledbackException)
                 tm.rollback();
```

Comment:
Giving that we used to handle ROLLING_BACK by calling tm.commit() and didn't have problems with that, I think your change to have ROLLING_BACK call tm.rollback(), is a safe change.  I prefer calling tm.rollback instead of tm.commit for ROLLING_BACK.